File: standardize_categories.js
const fs = require('fs');
const path = require('path');

const directory = 'c:\\Users\\Experttech.pk\\Desktop\\Dot-NK-E-Store\\frontend\\src\\pages\\ProductbyCategory';
const templateFile = path.join(directory, 'Mobiles.jsx');

if (!fs.existsSync(templateFile)) {
    console.error('Template file not found');
    process.exit(1);
}

const templateContent = fs.readFileSync(templateFile, 'utf8');
const idPattern = /\/api\/category\/(?:ies\/)?([a-f\d]{24})\/products/;
const componentNamePattern = /const\s+(\w+)\s+=\s+\(\{\s*categoryId\s*\}\)\s+=>\s+\{/;

const files = fs.readdirSync(directory).filter(f => f.endsWith('.jsx') && f !== 'Mobiles.jsx');

console.log(`Found ${files.length} files to standardize.`);

files.forEach(filename => {
    const filepath = path.join(directory, filename);
    const content = fs.readFileSync(filepath, 'utf8');

    const idMatch = content.match(idPattern);
    if (!idMatch) {
        console.log(`Skipping ${filename}: Could not find category ID`);
        return;
    }
    const categoryId = idMatch[1];

    let componentName;
    const nameMatch = content.match(componentNamePattern);
    if (!nameMatch) {
        componentName = filename.replace('.jsx', '');
    } else {
        componentName = nameMatch[1];
    }

    let collectionTitle = componentName.replace(/([a-z])([A-Z])/g, '$1 $2');
    if (!collectionTitle.endsWith('Collection')) {
        collectionTitle = `${collectionTitle} Collection`;
    }

    let newContent = templateContent;
    newContent = newContent.replace(/const Mobiles =/g, `const ${componentName} =`);
    newContent = newContent.replace(/export default Mobiles;/g, `export default ${componentName};`);
    newContent = newContent.replace(/\/api\/category\/[a-f\d]{24}\/products/g, `/api/category/${categoryId}/products`);
    newContent = newContent.split('Loading Mobile Collection...').join(`Loading ${collectionTitle}...`);
    newContent = newContent.split('Mobile Collection').join(collectionTitle);

    fs.writeFileSync(filepath, newContent, 'utf8');
    console.log(`Standardized: ${filename} (ID: ${categoryId}, Name: ${componentName})`);
});

console.log('All files processed.');

File: standardize_categories.py
import os
import re

directory = r"c:\Users\Experttech.pk\Desktop\Dot-NK-E-Store\frontend\src\pages\ProductbyCategory"
template_file = os.path.join(directory, "Mobiles.jsx")

with open(template_file, 'r', encoding='utf-8') as f:
    template_content = f.read()

# Define the pattern to find the category ID in the old files
# Usually looks like: /api/category/SOME_ID/products or /api/category/ies/SOME_ID/products
id_pattern = re.compile(r'/api/category/(?:ies/)?([a-f\d]{24})/products')

# Define the pattern to find the component name in the old files
component_name_pattern = re.compile(r'const\s+(\w+)\s+=\s+\(\{\s*categoryId\s*\}\)\s+=>\s+\{')

files = [f for f in os.listdir(directory) if f.endswith('.jsx') and f != "Mobiles.jsx"]

print(f"Found {len(files)} files to standardize.")

for filename in files:
    filepath = os.path.join(directory, filename)
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()
    
    # Extract Category ID
    id_match = id_pattern.search(content)
    if not id_match:
        print(f"Skipping {filename}: Could not find category ID")
        continue
    category_id = id_match.group(1)
    
    # Extract Component Name
    name_match = component_name_pattern.search(content)
    if not name_match:
        # Fallback: use filename without .jsx
        component_name = filename.replace('.jsx', '')
    else:
        component_name = name_match.group(1)
    
    # Create descriptive Title (e.g., "ActionFigures" -> "Action Figures Collection")
    # Simple camelCase to Space Case
    collection_title = re.sub(r'([a-z])([A-Z])', r'\1 \2', component_name)
    if not collection_title.endswith("Collection"):
        collection_title = f"{collection_title} Collection"

    # Generate new content based on template
    new_content = template_content
    
    # Replacement 1: Component Name definition
    new_content = re.sub(r'const Mobiles =', f'const {component_name} =', new_content)
    
    # Replacement 2: export default
    new_content = re.sub(r'export default Mobiles;', f'export default {component_name};', new_content)
    
    # Replacement 3: Category ID in useEffect
    # Template has: `/api/category/6929e21f9ee403bf655019b4/products`
    new_content = re.sub(r'/api/category/[a-f\d]{24}/products', f'/api/category/{category_id}/products', new_content)
    
    # Replacement 4: Page Titles and Loader text
    new_content = new_content.replace('Loading Mobile Collection...', f'Loading {collection_title}...')
    new_content = new_content.replace('Mobile Collection', collection_title)
    
    # Replacement 5: Specific emoji/text if any (Template has 📵)
    # We can keep it or try to be smart, but standardizing on 📵 is fine if it's the new style.
    
    with open(filepath, 'w', encoding='utf-8') as f:
        f.write(new_content)
    
    print(f"Standardized: {filename} (ID: {category_id}, Name: {component_name})")

print("All files processed.")
